### ISSUE TITLE
CFamily - cancel in-process analysis if a new analysis is triggered for the same file

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#1309]
### Description
It's possible to trigger a new analysis before the previous analysis has been finished. The previous analysis execution is not cancelled, and both analysis run in parallel.
This might lead to 2 problems:
1. Issue correctness: if the first analysis takes longer than the second analysis, we might display issues that are no longer relevant.
2. Machine resources: the first analysis is redundant and consumes resources.

### Proposed solution
Once a new request is made, cancel the previous one. We can queue the analysis requests in a thread-safe queue, based on file name. 

## Testing scenarios
- First analysis has not started yet
- First analysis has started, but not yet finished
- First analysis has finished, but issues not yet reported

## Notes
- Should be tested in VS 2015 / 2017 / 2019 due to different threading models.
